code/modules/paperwork/paper_cutter.dm

```diff
@@ -6,108 +6,161 @@
+/obj/item/papercutter/attackby(obj/item/inserted_item, mob/user, params)
+	if(istype(inserted_item, /obj/item/paper) && !istype(inserted_item, /obj/item/paper/paperslip))
+		if(stored_paper)
+			balloon_alert(user, "already paper inside!")
+		if(!user.transferItemToLoc(inserted_item, src))
 			return
 		playsound(loc, SFX_PAGE_TURN, 60, TRUE)
-		to_chat(user, span_notice("You place [P] in [src]."))
-		storedpaper = P
-		update_appearance()
-		return
-	if(istype(P, /obj/item/hatchet/cutterblade) && !storedcutter)
-		if(!user.transferItemToLoc(P, src))
+		balloon_alert(user, "paper inserted")
+		stored_paper = inserted_item
+
+	if(istype(inserted_item, /obj/item/hatchet/cutterblade))
+		if(stored_blade)
+			balloon_alert(user, "already a blade inside!")
 			return
-		to_chat(user, span_notice("You replace [src]'s [P]."))
-		P.forceMove(src)
-		storedcutter = P
-		update_appearance()
-		return
+		if(!user.transferItemToLoc(inserted_item, src))
+			return
+		balloon_alert(user, "blade inserted")
+		inserted_item.forceMove(src)
+		stored_blade = inserted_item
+
+	update_appearance()
+
 	..()
 
 /obj/item/papercutter/attack_hand(mob/user, list/modifiers)
-	. = ..()
 	if(.)
 		return
 	add_fingerprint(user)
-	if(!storedcutter)
-		to_chat(user, span_warning("The cutting blade is gone! You can't use [src] now."))
-		return
 
-	if(!cuttersecured)
-		to_chat(user, span_notice("You remove [src]'s [storedcutter]."))
-		user.put_in_hands(storedcutter)
-		storedcutter = null
+	if(!stored_blade && stored_paper)
+		balloon_alert(user, "no blade!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	else if(!blade_secured)
+		balloon_alert(user, "blade removed")
+		user.put_in_hands(stored_blade)
+		stored_blade = null
 		update_appearance()
+		return COMPONENT_CANCEL_ATTACK_CHAIN
 
-	if(storedpaper)
-		playsound(src.loc, 'sound/weapons/slash.ogg', 50, TRUE)
-		to_chat(user, span_notice("You neatly cut [storedpaper]."))
-		storedpaper = null
-		qdel(storedpaper)
-		new /obj/item/paper/paperslip(get_turf(src))
-		new /obj/item/paper/paperslip(get_turf(src))
+	else if(stored_paper)
+		balloon_alert(user, "paper removed")
+		user.put_in_hands(stored_paper)
+		stored_paper = null
 		update_appearance()
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	// If there's a secured blade but no paper, just pick it up
+	. = ..()
+
+/obj/item/papercutter/attack_hand_secondary(mob/user, list/modifiers)
+	if(!stored_blade)
+		balloon_alert(user, "no blade!")
+	else if(!blade_secured)
+		balloon_alert(user, "blade unsecured!")
+	else if(!stored_paper)
+		balloon_alert(user, "nothing to cut!")
+	else
+		cut_paper(user)
+
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/papercutter/proc/cut_paper(mob/user)
+	var/clumsy = (iscarbon(user) && HAS_TRAIT(user, TRAIT_CLUMSY) && prob(cut_self_chance))
+	playsound(src.loc, 'sound/weapons/slash.ogg', 50, TRUE)
+	if(clumsy)
+		to_chat(user, span_userdanger("You neatly cut [stored_paper][clumsy ? "... and your finger in the process!" : "."]"))
```

Comment:
```suggestion
	playsound(src.loc, 'sound/weapons/slash.ogg', 50, TRUE)
	to_chat(user, span_userdanger("You neatly cut [stored_paper][clumsy ? "... and your finger in the process!" : "."]"))
	if(clumsy)
```

Otherwise we would never ever get a to_chat for non-clumsy people ever